src/ObjectHelper.php

```diff
@@ -13,9 +13,17 @@ class ObjectHelper
     public function buildTransaction(stdClass $object)
     {
         $customer = isset($object->account) ? $this->buildCustomer($object->account) : null;
-        $paymentMethod = isset($object->sourcePaymentMethod)
-                       ? $this->buildPaymentMethod($object->sourcePaymentMethod)
-                       : null;
+
+        // fetch transaction response may have salesTaxAddress field as shipping address
+        // we should dump this info into the card and use it when creating transaction record
+        $paymentMethod = null;
+        if (isset($object->sourcePaymentMethod)) {
+            if (isset($object->salesTaxAddress)) {
+                $paymentMethod = $this->buildPaymentMethod($object->sourcePaymentMethod, $object->salesTaxAddress);
+            } else {
+                $paymentMethod = $this->buildPaymentMethod($object->sourcePaymentMethod);
+            }
```

Comment:
I think you can make this simple if you store $object->salesTaxAddress in a variable
something like 
```php
$sales_tax_address = isset($object->salesTaxAddress) ? $object->salesTaxAddress : null;
$paymentMethod = $this->buildPaymentMethod($object->sourcePaymentMethod, $sales_tax_address);
```